working/333 - shared memory multithreading/proposal.md

```diff
@@ -1195,7 +855,7 @@ to control threads.
 ```dart
 // dart:concurrent
 
-abstract class Thread implements Shareable {
+abstract class Thread {
```

Comment:
The below:
> /// ... which uses the
>  /// spawned thread as an executor for all callbacks: this
>  /// means the thread will remain alive as long as there is
>  /// a callback referencing it.

I'm not sure what that means. Are the callbacks `Zone`-functions, like `scheduleMicrotask`, or are they the callback arguments to those zone functions?

Are executors part of zones, or separator?
(If zones are shareable, and you can share the value with a function run with a different `Executor.schedule`, then it seems the executor is not bound by the zone or vice-versa.)

Reply:
I think what this comment is trying to say is that if you have some sort of branching asynchronous execution (e.g. you do `future.then(cb)`) then `cb` will run on the spawned thread. 

Maybe this is just bad API: if this behavior is desired users should use executors explicitly.